Documentation/contributing.rst

```diff
@@ -55,6 +55,39 @@ requirements have been met:
 * You have signed off on your commits, see the section "Developer's
   Certificate of Origin" for more details.
 
+Release Process
+---------------
+
+Cilium schedules a major release every 3 months. Each major release is
+performed by incrementing the `Y` in the version format `X.Y.0`. The group of
+committers can decide to increment `X` instead to mark major milestones in
+which case `Y` is reset to 0.
+
+The following steps are performed to publish a release:
+
+1. The master branch is set to the version `X.Y.90` at all times. This ensures
```

Comment:
F.e. in netsniff-ng project, we use `git describe --always` scheme in a generated header file during build, so that it outputs a development version after the tag like `v0.6.2-48-g88ec2ad`, which can then be used in user reports for checking things like `git log`, could be another option (but would require git for development build)?

Anyway, not an expert in package management, for the `+` I just checked debian package management and the versioning information is part of the `changelog` file if I understand it correctly (https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version, part `upstream_version`), and it's supported there. There's also a `watch` file that contains a base url plus defineable regex to let debian check for new releases at some location.

The `Version:` field in Fedora would support it (https://docs.fedoraproject.org/en-US/Fedora_Draft_Documentation/0.1/html/Packagers_Guide/chap-Packagers_Guide-Spec_File_Reference-Preamble.html).